module/Application/src/Application/View/Helper/ScriptUrl.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Application\View\Helper;
+
+use Zend\View\Helper\AbstractHelper;
+use Zend\View\Exception;
+
+/**
+ * Url view helper for use inside javascript code.
+ * Usage: $this->scriptUrl()->requireUrl('/url/route');
+ *
+ * @package Application\View\Helper
+ */
+class ScriptUrl extends AbstractHelper
+{
+    /**
+     * @return \Application\View\Helper\ScriptUrl
+     */
+    public function __invoke()
+    {
+        return $this;
+    }
+
+    /**
+     * Makes an url route available to the javascript url helper.
+     *
+     * @param string $name Name of the route.
+     * @param array $params List of route parameters to make available.
+     *
+     * @return \Application\View\Helper\ScriptUrl
+     */
+    public function requireUrl($name, $params = array())
+    {
+        $scriptParams = array();
+
+        foreach($params as $param) {
+            $scriptParams[$param] = '{' . $param . '}';
+        }
+
+        $url = $this->getView()->url($name, $scriptParams);
+        // Include data as inline script
+        $this->getView()->inlineScript()->captureStart();
+        echo 'URLHelper.addUrl("'. $name . '", "'. urldecode($url) . '");';
```

Comment:
This violates the SRP. A PHP either defines logic, or it defines output. This file now does both. In this case there is a hidden dependency between the PHP and JavaScript (the JavaScript URLHelper object), which will be difficult to find since it is not in the template files.  Line 42 - 44 need to be moved to the template files, and the helper should keep an array of all the urls